tests/pytest_tests/system_tests/test_sweep/test_sweep_scheduler.py

```diff
@@ -774,6 +775,6 @@ async def mock_launch_add(*args, **kwargs):
     scheduler = SweepScheduler(
         api, sweep_id=s, entity=user, project="t", queue="q", num_workers=1
     )
-    scheduler._register_agents()
+    asyncio.run(scheduler._register_agents())
```

Comment:
because this is a private method, i think this is fine. if it were public id like to have the async part handled internally...